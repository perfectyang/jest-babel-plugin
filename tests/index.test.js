const trans = require('../task/trans.js');
const babel = require("@babel/core");

describe('block', () => {
  const input = `
    const a = 'perfectyang'
    const b = 'goood'
  `
  it('dev', () => {
    const {code} = babel.transform(input, {plugins: [
      [
       trans,
       {
        options: [
          {
            name: 'perfectyang'
          }
        ]
       }
      ]
    ]})
    expect(code).toMatchSnapshot();
  });
  // it('pro', () => {
  //   const {code} = babel.transform(input, {plugins: [
  //     [
  //      trans,
  //      {
  //        name: 'perfectyang'
  //      }
  //     ]
  //   ]})
  //   expect(code).toMatchSnapshot();
  // });

})
