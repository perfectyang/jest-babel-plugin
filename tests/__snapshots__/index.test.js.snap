// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block dev 1`] = `
"\\"use strict\\";

const a = 'perfectyang';
const b = 'goood';"
`;
